staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go

```diff
@@ -27,27 +27,46 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apiserver/pkg/authentication/authenticator"
 	"k8s.io/apiserver/pkg/endpoints/handlers/responsewriters"
-	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	apirequest "k8s.io/apiserver/pkg/endpoints/request"
 	"k8s.io/klog/v2"
 )
 
+// NoopCheck is used for WithAuthentication calls outside of the apiserver,
+// where an apiSelfRequestCheck function is not needed
+func NoopCheck(req *http.Request) bool {
+	return false
+}
+
 // WithAuthentication creates an http handler that tries to authenticate the given request as a user, and then
 // stores any such user found onto the provided context for the request. If authentication fails or returns an error
 // the failed handler is used. On success, "Authorization" header is removed from the request and handler
 // is invoked to serve the request.
-func WithAuthentication(handler http.Handler, auth authenticator.Request, failed http.Handler, apiAuds authenticator.Audiences) http.Handler {
+func WithAuthentication(handler http.Handler, auth authenticator.Request, failed http.Handler, apiAuds authenticator.Audiences, apiSelfRequestCheck func(*http.Request) bool, requestInfoFactory *apirequest.RequestInfoFactory) http.Handler {
```

Comment:
When I said the authentication stuff knows this, I was not expecting you to modify this method, I actually don't recommend this, sorry for not being clear!

Instead, I was expecting a wrapper around the specific authenticator that handles the self requests, that passes through requests and increments the metric if that authenticator returns true.